envoy/router/path_match.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Decides if the target route path matches the provided pattern.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathMatcher : Logger::Loggable<Logger::Id::router> {
+public:
+  PathMatcher() = default;
+  virtual ~PathMatcher() = default;
+
+  /**
+   * Used to determine if the current url matches the predicate pattern.
+   *
+   * @param url current url of route
+   * @return true if route url matches the predicate pattern.
+   */
+  virtual bool match(absl::string_view path) const PURE;
+
+  /**
+   * @return the match pattern of the predicate.
```

Comment:
These comments talk about "predicate" but I'm not quite sure what that means. The class is call "path matcher" not "path match predicate". Perhaps this comment should say: `@return the pattern for this matcher`. (That being said, what is the format of this returned value? Is it in regex syntax? Please document)